src/utils.js

```diff
@@ -58,6 +58,33 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  variables.forEach((variable) => {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+    }
+  });
+
+  if (typeof scopeVar === 'undefined' ||
+      typeof scopeVar.defs === 'undefined' ||
+      typeof scopeVar.defs[0] === 'undefined' ||
```

Comment:
Honestly I'm not sure, I _think_ that `defs` holds all definitions of the variable starting at the most locally scoped version but again, I don't know. I haven't been able to find docs talking about this.